packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/scheduleWalkthrough.ts

```diff
@@ -0,0 +1,166 @@
+import inquirer from 'inquirer';
+import path from 'path';
+import fs from 'fs-extra';
+import { FunctionParameters } from 'amplify-function-plugin-interface';
+inquirer.registerPrompt('datetime', require('inquirer-datepicker'));
+import { CronBuilder } from '../utils/cronBuilder';
+import { constructCloudWatchEventComponent } from '../utils/cloudformationHelpers';
+import { minuteHelper, hourHelper, timeHelper, weekHelper, monthHelper, yearHelper } from '../utils/cronHelper';
+import { CronExpressionsMode } from '../utils/constants';
+import { CronExpression } from '../utils/cronExpression';
+const categoryName = 'function';
+
+export async function scheduleWalkthrough(
+  context: any,
+  params: Partial<FunctionParameters>,
+): Promise<Pick<FunctionParameters, 'cloudwatchRule'>> {
+  const projectBackendDirPath = context.amplify.pathManager.getBackendDirPath();
+  const resourceDirPath = path.join(projectBackendDirPath, categoryName, params.resourceName);
+  const cfnFileName = `${params.resourceName}-cloudformation-template.json`;
+  const cfnFilePath = path.join(resourceDirPath, cfnFileName);
+  let scheduleParams = params;
+  if (params.cloudwatchRule === undefined || params.cloudwatchRule === 'false') {
+    if (await context.amplify.confirmPrompt.run('Do you want to invoke this function on a recurring schedule?', false)) {
+      try {
+        let cloudWatchRule = await cronServiceWalkthrough(context);
```

Comment:
be consistent about cloudwatchRule vs cloudWatchRule. I think I would write it cloudwatchRule because "cloudwatch" is one entity and "rule" is a modifier on that entity. But whichever you choose, just be consistent.